balancer-js/src/test/lib/joinHelper.ts

```diff
@@ -32,7 +33,14 @@ export const testExactTokensIn = async (
   expect(transactionReceipt.status).to.eq(1);
   expect(BigInt(expectedBPTOut) > 0).to.be.true;
   const expectedDeltas = insert(amountsIn, pool.bptIndex, expectedBPTOut);
-  expect(expectedDeltas).to.deep.eq(balanceDeltas.map((a) => a.toString()));
+  if (approximate) {
+    console.log(`!!!!!!! APPROX TEST`);
+    const diffs = expectedDeltas.map((e, i) =>
+      BigNumber.from(e).sub(balanceDeltas[i]).abs()
+    );
+    diffs.forEach((diff) => expect(diff.lt('100000000000000000')).to.be.true);
+  } else
```

Comment:
Maybe a better approach here would be to use the `accuracy` helper to check for relative rather than absolute approximation.
Not sure if that's something worth spending more time though.